lcc_comchain_base/models/company.py

```diff
@@ -19,3 +26,54 @@ def _default_messages(self):
     comchain_odoo_wallet_password = fields.Char(string="Odoo wallet password")
     message_from = fields.Char("Message from", default=_default_messages)
     message_to = fields.Char("Message to", default=_default_messages)
+
+    last_block_checked_nb = fields.Integer('Last Block Checked Nb')
+
+    def _retrieve_last_debit_transactions(self):
+        """A list of transactions (dictionnary) is expected, with the following data:
+        - sender: the Odoo name of the wallet concerned by the debit request,
+        - amount: the amount debited from the wallet,
+        - tx_id: the transaction ID in the digital currency backend
+        - tx_timestamp: the timestamp of the transaction
+        """
+        res = super(Company, self)._retrieve_last_debit_transactions()
+
+        # Retrieve all the debit transactions from the newly created blocks
+        company_id = self.env.user.company_id
+        try:
+            # Get the ID of the last blockchain block built
+            last_block_id = pyc3l.getBlockNumber()
+            # Loop on all the blocks created since the last check
+            _logger.info("Start reconversion check on Comchain from block %s to block %s" % (company_id.last_block_checked_nb, last_block_id) )
+            for block_nb in range(company_id.last_block_checked_nb + 1, last_block_id + 1):
+                _logger.info("Get Debit Transactions - read block %s" % block_nb)
+                # Get the block transactions
+                block_txs = pyc3l.BlockByNumber(block_nb).bc_txs
+                for tx in block_txs:
+                    if not tx.currency:
+                        continue # this transaction is not linked with a currency
+                    if not tx.currency.name.lower() == company_id.comchain_currency_name.lower():
+                        continue # this transaction is about another smart contract
+                    full_tx = tx.full_tx
+                    if not full_tx.is_cc_transaction:
+                        continue # this transaction is not a digital currency transfer
+                    if not full_tx.direction == 1:
+                        continue # this transaction is not a reconversion
+                    _logger.info("Transactions %s retrieved" % tx.hash)
+                    sender_adress = full_tx.add2.lstrip("0x")
+                    matched_wallet = self.env["res.partner.backend"].search([("comchain_id", "ilike", sender_adress)])
+                    if len(matched_wallet) == 0:
+                        continue # this transactions comes from a wallet no registered in current Odoo database
+                    string_amount = str(full_tx.sent)
+                    res.append({
+                        "sender": "comchain:%s" % sender_adress,
+                        "amount": f"{string_amount[0:-2]}.{string_amount[-2:]}",
+                        "tx_id": tx.hash,
+                        "tx_timestamp": datetime.fromtimestamp(int(full_tx.receivedat["value"])),
```

Comment:
This value doesn't mean anything. If your version of pyc3l supports it, try `full_tx.received_at` (which is a datetime already). Otherwise you'll also probably have a wrong datetime (it is an utc timestamp, and you should store timezone aware datetime...)

Reply:
fixed by using directly full_tx.received_at